app/src/main/java/com/android/streetworkapp/model/event/EventRepositoryFirestore.kt

```diff
@@ -68,6 +68,62 @@ class EventRepositoryFirestore(private val db: FirebaseFirestore) : EventReposit
     }
   }
 
+  /**
+   * Add a participant to an event.
+   *
+   * @param eid The event ID.
+   * @param uid The user ID.
+   */
+  override suspend fun addParticipantToEvent(eid: String, uid: String) {
+    require(eid.isNotEmpty())
+    require(uid.isNotEmpty())
+    try {
+      val event = getEventByEid(eid)
+      if (event != null) {
+        val updatedParticipants = event.participants + 1
+        val updatedListParticipants = event.listParticipants.toMutableList()
+        updatedListParticipants.add(uid)
+        db.collection(COLLECTION_PATH)
+            .document(eid)
+            .update(
+                "participants", updatedParticipants, "listParticipants", updatedListParticipants)
```

Comment:
Improvement: `updatedParticipants` can lead to race conditions (if multiple clients try to update the event at the same time) instead I discover the `FieldValue.increment(1)`. With this change you can remove the line 83.

Reply:
good call! this has been changed